.github/workflows/e2e-pool.yml

```diff
@@ -0,0 +1,232 @@
+name: Testing
+
+on: [pull_request]
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prepare_variables:
+    runs-on: ubuntu-22.04
+    outputs:
+      POOL_NAME: ${{ steps.generate.outputs.POOL_NAME }}
+      POOL_INSTANCE: ${{ steps.generate.outputs.POOL_INSTANCE }}
+      BASE_URL: ${{ steps.generate.outputs.BASE_URL }}
+      API_URI: ${{ steps.generate.outputs.API_URI }}
+
+    steps:
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v4
+
+      - name: Generate
+        id: generate
+        run: |
+          echo "BASE_URL=${{ env.GITHUB_HEAD_REF_SLUG_URL }}.dashboard.saleor.rocks" >> $GITHUB_OUTPUT
+          echo "API_URI=https://${{ env.GITHUB_HEAD_REF_SLUG_URL }}.staging.saleor.cloud/graphql/" >> $GITHUB_OUTPUT
+          echo "POOL_NAME=${{ env.GITHUB_HEAD_REF_SLUG_URL }}" >> $GITHUB_OUTPUT
+          echo "POOL_INSTANCE=https://${{ env.GITHUB_HEAD_REF_SLUG_URL }}.staging.saleor.cloud" >> $GITHUB_OUTPUT
+
+  prepare_instance:
+    runs-on: ubuntu-22.04
+    needs: prepare_variables
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v4
+
+      - name: Write config file
+        id: write-config-file
+        env:
+          ACCESS_TOKEN: ${{ secrets.STAGING_TOKEN }}
+        run: echo '{"token":"Token ${{ env.ACCESS_TOKEN }}","telemetry":"false","organization_slug":"saleor","organization_name":"Saleor"}' > ~/.config/saleor.json
```

Comment:
This could lead to `ACCESS_TOKEN` leak as it's logged to the console.
It's only thanks to GitHub Actions secrets masking it wasn't revealed.